### ISSUE TITLE
Field "deliveryMasterId" and "deliveryMaster.id" collide for orderBy enum generation

### ISSUE THREAD

**Issue by kroeder** [dotansimha/graphql-code-generator#991]
**Describe the bug**
If you have something like this
```
Shipment {
  deliveryMasterId: string;
  deliveryMaster: {
    id: string;
  }
}
```
then the generator generates colliding code: (I think this is generated by typescript-common)
```
ShipmentDeliveryMasterIdAsc = "shipment_deliveryMaster_id_ASC",
ShipmentDeliveryMasterIdAsc = "shipment_deliveryMasterId_ASC",
```

~This is kinda error-by-schema-design and I'm in a discussion if we want our schema that way but  I'm also not sure if this can happen in other scenarios so I just wanted to let you know that this does not work currently~ We actually need this and it makes sense that we have our schema that way.

**To Reproduce**
I don't have a public schema, let me know if you need one. But it will end up like my example at the top of this post

**Expected behavior**
I'm not sure. Maybe add an option where you allow to underscore every graph-navigation?
```
Shipment_DeliveryMasterId
Shipment_DeliveryMaster_Id
```

**Additional info**
If there's a good solution or if you are fine with an underscore option I'd be happy to contribute if this speeds up the process! It might be not a high priority for you but is unfortunately a show stopper for us internally

**Environment:**
 - OS: Windows 10
 - Codegen: 0.14.5
 - Node: 10.x.x

**Comment by dotansimha** [comment]
Fixed in the recent refactor, will release soon.

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍
